File: eval/fixtures/php-taskflow/src/Store/MemStore.php
<?php

declare(strict_types=1);

namespace App\Store;

/** In-memory Store implementation; assigns ids via the Sequence trait. */
class MemStore implements Store
{
    use Sequence;

    /** @var array<int,Task> */
    private array $tasks = [];

    public function get(int $id): ?Task
    {
        return $this->tasks[$id] ?? null;
    }

    /** @return Task[] */
    public function all(): array
    {
        return array_values($this->tasks);
    }

    public function add(Task $task): void
    {
        if ($task->id === 0) {
            $task->id = $this->nextId();
        }
        $this->tasks[$task->id] = $task;
    }

    public function complete(int $id): void
    {
        $task = $this->get($id);
        if ($task === null) {
            throw NotFoundError::forId($id);
        }
        $task->markDone();
    }

    public function remove(int $id): void
    {
        if (!isset($this->tasks[$id])) {
            throw NotFoundError::forId($id);
        }
        unset($this->tasks[$id]);
    }
}

File: eval/fixtures/python-taskflow/taskflow/store/memstore.py
"""In-memory Store implementation; assigns ids via the Sequence mixin."""

from typing import Dict, List, Optional

from taskflow.store.base import Store
from taskflow.store.errors import NotFoundError
from taskflow.store.sequence import Sequence
from taskflow.store.task import Task


class MemStore(Sequence, Store):
    def __init__(self) -> None:
        super().__init__()
        self._tasks: Dict[int, Task] = {}

    def get(self, id: int) -> Optional[Task]:
        return self._tasks.get(id)

    def all(self) -> List[Task]:
        return list(self._tasks.values())

    def add(self, task: Task) -> None:
        if task.id == 0:
            task.id = self.next_id()
        self._tasks[task.id] = task

    def complete(self, id: int) -> None:
        task = self.get(id)
        if task is None:
            raise NotFoundError.for_id(id)
        task.mark_done()

    def remove(self, id: int) -> None:
        if id not in self._tasks:
            raise NotFoundError.for_id(id)
        del self._tasks[id]
